src/search_unused.rs

```diff
@@ -769,6 +769,48 @@ fn test_crate_renaming_works() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[test]
+fn test_unused_renamed_in_registry() -> anyhow::Result<()> {
+    // when a lib like xml-rs is exposed with a different name,
+    // cargo-machete reports the unused spec properly.
+    let analysis = find_unused(
+        &PathBuf::from(TOP_LEVEL).join("./integration-tests/unused-renamed-in-registry/Cargo.toml"),
+        UseCargoMetadata::Yes,
+    )?
+    .expect("no error during processing");
+    assert_eq!(analysis.unused, &["xml-rs".to_string()]);
+
+    Ok(())
+}
+
+#[test]
+fn test_unused_renamed_in_spec() -> anyhow::Result<()> {
+    // when a lib is renamed through key = { package = … },
+    // cargo-machete reports the unused spec properly.
+    let analysis = find_unused(
+        &PathBuf::from(TOP_LEVEL).join("./integration-tests/unused-renamed-in-spec/Cargo.toml"),
+        UseCargoMetadata::Yes,
+    )?
+    .expect("no error during processing");
+    assert_eq!(analysis.unused, &["tracing".to_string()]);
+
+    Ok(())
+}
+
+#[test]
+fn test_unused_kebab_spec() -> anyhow::Result<()> {
+    // when a lib is renamed through key = { package = … },
+    // cargo-machete reports the unused spec properly.
```

Comment:
Think this comment is a copy-pasto from the previous one, and should be updated maybe?